perceval/components/linear_circuit.py

```diff
@@ -537,7 +500,7 @@ def copy(self, subs: Union[dict,list] = None):
     def decomposition(U: MatrixN,
```

Comment:
Don't we want this method to explicitly fail instead of returning a "None" ?
Consider adding a warning at the end of the method in this case "cannot decompose this circuit ..."